notify-on-job-start

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -euo pipefail
+
+: "${SLACK_TOKEN:?The SLACK_TOKEN environment variable is required.}"
+: "${SLACK_CHANNELS:?The SLACK_CHANNELS environment variable is required.}"
+
+: "${AWS_BATCH_JOB_ID:=}"
+: "${GITHUB_RUN_ID:=}"
+
+bin="$(dirname "$0")"
+job_name="${1:?A job name is required as the first argument}"
+repo_name="${2:?A repository name is required as the second argument}"
```

Comment:
*(non-blocking)*

Although in the short term it's unlikely that external repos will be using these scripts, I like to think of GitHub repos by org/name pair to make it more generalizable.

```suggestion
github_repo="${2:?A GitHub repository is required as the second argument}"
```

References would then be updated to something like:

```diff
- message+=" The job was submitted by GitHub Action <https://github.com/nextstrain/${repo_name}/actions/runs/${GITHUB_RUN_ID}?check_suite_focus=true|${GITHUB_RUN_ID}>."
+ message+=" The job was submitted by GitHub Action <https://github.com/${github_repo}/actions/runs/${GITHUB_RUN_ID}?check_suite_focus=true|${GITHUB_RUN_ID}>."
```

```diff
- message+=" Follow along in your local \`${repo_name}\` repo with: "'```'"nextstrain build --aws-batch --no-download --attach ${AWS_BATCH_JOB_ID} ${build_dir}"'```'
+ message+=" Follow along in your local clone of ${github_repo} with: "'```'"nextstrain build --aws-batch --no-download --attach ${AWS_BATCH_JOB_ID} ${build_dir}"'```'
```

Same for `notify-on-job-fail`.

Reply:
Great suggestion! Updated in https://github.com/nextstrain/ingest/commit/b2a0de70342cef1b2b27f3f139af262910466b24